### ISSUE TITLE
Adding EKS Nodepool reports error, but really succeds

### ISSUE THREAD

**Issue by doddatpivotal** [vmware/terraform-provider-tanzu-mission-control#202]
### Describe the bug

I provisioned an EKS cluster using TMC Terraform Provider 1.1.8.  I then went and added a second nodepool to the cluster and applied the changes.  After about a minute, I got this error...

```tanzu-mission-control_ekscluster.eks_1: Still modifying... [id=c:01H5WDQF5C6QHH9GD72QC4ABWS, 1m20s elapsed]
╷
│ Error: Unable to update Tanzu Mission Control EKS cluster entry, name : eks-1: Unable to update Tanzu Mission Control EKS cluster entry, name : eks-1: PUT request failed with status : 400 Bad Request, response: {"error":"nothing to update","code":3,"message":"nothing to update"}
│ 
│   with tanzu-mission-control_ekscluster.eks_1,
│   on main.tf line 50, in resource "tanzu-mission-control_ekscluster" "eks_1":
│   50: resource "tanzu-mission-control_ekscluster" "eks_1" {
│ 
```

However, after that, I checked TMC and the nodepool was added successfully.  I then ran terraform apply again, and it showed no changes and I was up to date.
[adding-node-pool.log](https://github.com/vmware/terraform-provider-tanzu-mission-control/files/12131706/adding-node-pool.log)

After my first failure, I wanted to see if I could reproduce it, so I added a 3rd node pool.  The same series of events happened.  Terraform apply failed with same error message, and then everything worked.  Attaching complete command output 

### Reproduction steps

Captured above.

### Expected behavior

Terraform apply to show success.

**Comment by asutoshpalai** [comment]
Hi, is this bug still present in the latest version? I think I had seen this error recently and sent in a fix with #183.

**Comment by doddatpivotal** [comment]
Just tested on v1.2 and did not face any issues.
